test/unit/ui/test_solara.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from pathlib import Path
+
+import ipywidgets
+import solara
+from solara.server import reload
+from solara.server.app import AppScript
+
+logger = logging.getLogger("solara.server.app_test")
+
+HERE = Path(__file__).parent
+reload.reloader.start()
+
+
+def app_box_and_rc(app_name, kernel_context):
+    app = AppScript(app_name)
+    app.init()
+    try:
+        with kernel_context:
+            el = app.run()
+            root = solara.RoutingProvider(
+                children=[el], routes=app.routes, pathname="/"
+            )
+            box, rc = solara.render(root, handle_error=False)
+            yield box, rc
+    finally:
+        app.close()
+
+
+def test_notebook_widget(kernel_context, no_kernel_context):
```

Comment:
`no_kernel_context` is not used here. Is it still needed? Why? 
If not:
* Do we need the fixture at all? 
* If yes: Could you please add an expressive tests using the fixture?